shell/browser/api/electron_api_data_pipe_holder.cc

```diff
@@ -103,39 +103,19 @@ class DataPipeReader {
   }
 
   void OnSuccess() {
-    // Pass the buffer to JS.
-    //
-    // Note that the lifetime of the native buffer belongs to us, and we will
-    // free memory when JS buffer gets garbage collected.
+    // Copy the buffer to JS.
+    // TODO(nornagon): make this zero-copy by allocating the array buffer
```

Comment:
Does  https://github.com/electron/electron/pull/34403/commits/47674d53611180d27430f3bb07ba07e428ce0fae accomplish this so that the TODO can be removed? Or is 4767 a prerequisite? (Or neither? :dizzy_face:)

Reply:
no, this still copies. to fix this we need to allocate the buffer we stream into using the v8 arraybuffer allocator, which is a bit more of a refactor.